### DISCUSSION TITLE
Do not audit when only ignored columns are updated

### DISCUSSION THREAD

**Discussion by reschmann** [thepirat000/Audit.NET#413] [Q&A]
I have a table with some columns set [AuditIgnore] attribute. But when a row is updated and only ignored columns are modified with new value, audit is saved with "no columns changes". 

In that case I would like to not audit at all when only ignored columns are updated.
Something like cancel AuditEntityAction when entry.Action == "Update" && entry.Changes.Count == 0

**Comment by thepirat000** [comment]
You can do that with a custom action.

For example, on your startup code:

```c#
Audit.Core.Configuration.AddCustomAction(ActionType.OnScopeCreated, scope =>
{
    var efEvent = scope.GetEntityFrameworkEvent();
    if (efEvent != null && efEvent.Entries.All(e => e.Changes.Count == 0))
    { 
        scope.Discard();
    }
});
```
